pkg/models/terminal/terminal.go

```diff
@@ -224,3 +233,84 @@ func (t *terminaler) HandleSession(shell, namespace, podName, containerName stri
 
 	session.Close(1, "Process exited")
 }
+
+func (t *terminaler) HandleShellAccessToNode(nodename string, conn *websocket.Conn) {
+	namespace, podName, containerName, shell, privileged, labels := "kubesphere-controls-system", nodename+"-shell-access", "nsenter", "sh", true, map[string]string{"shell-access": nodename}
+
+	pod, err := t.client.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+
+	if err != nil || (pod.Status.Phase != "Running" && pod.Status.Phase != "Pending") {
+		var p = &v1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: labels,
+			},
+			Spec: v1.PodSpec{
+				NodeName:      nodename,
+				HostPID:       true,
+				HostNetwork:   true,
+				RestartPolicy: v1.RestartPolicyNever,
+				Containers: []v1.Container{
+					{
+						Name:  containerName,
+						Image: "alexeiled/nsenter:latest",
+						Command: []string{
+							"/nsenter", "-m", "-u", "-i", "-n", "-p", "-t", "1", "tail", "-f", "/dev/null",
+						},
+						Stdin: true,
+						TTY:   true,
+						SecurityContext: &v1.SecurityContext{
+							Privileged: &privileged,
+						},
+					},
+				},
+			},
+		}
+
+		pod, err = t.client.CoreV1().Pods(namespace).Create(context.Background(), p, metav1.CreateOptions{})
+		if err != nil {
+			klog.Warning(err)
+			return
+		}
+	}
+
+	t.WatchPodStatusBeRunning(pod, func() {
+		mu.Lock()
+		if _, ok := NodeSessionCounter[nodename]; !ok {
+			var index int64 = 0
+			NodeSessionCounter[nodename] = &index
+		}
+		mu.Unlock()
+
+		atomic.AddInt64(NodeSessionCounter[nodename], 1)
+		t.HandleSession(shell, namespace, podName, containerName, conn)
```

Comment:
Do we need to add a timeout logic to each session?

Reply:
because I can't detect whether the session is active, it is hard to add timeout logic. If this is necessary, an active detection mechanism needs to be added to complete this function.

Reply:
> because I can't detect whether the session is active, it is hard to add timeout logic. If this is necessary, an active detection mechanism needs to be added to complete this function.

OK. Let's talk about this topic tomorrow online. And welcome to share!

Reply:
> > because I can't detect whether the session is active, it is hard to add timeout logic. If this is necessary, an active detection mechanism needs to be added to complete this function.
> 
> OK. Let's talk about this topic tomorrow online. And welcome to share!

Maybe we need to list all nodes and pre-set the nodename to this map ?

Reply:
OK, but when considering this function, we also need to consider the FE code. Because when the connection is disconnected, the FE will be automatically reconnected